doc/source/area-detector.rst

```diff
@@ -269,6 +269,168 @@ The vendor specific details are embedded in the cams
 .. inheritance-diagram:: ophyd.areadetector.cam.CamBase ophyd.areadetector.cam.AdscDetectorCam ophyd.areadetector.cam.Andor3DetectorCam ophyd.areadetector.cam.AndorDetectorCam ophyd.areadetector.cam.BrukerDetectorCam ophyd.areadetector.cam.FirewireLinDetectorCam ophyd.areadetector.cam.FirewireWinDetectorCam ophyd.areadetector.cam.LightFieldDetectorCam ophyd.areadetector.cam.Mar345DetectorCam ophyd.areadetector.cam.MarCCDDetectorCam ophyd.areadetector.cam.PSLDetectorCam ophyd.areadetector.cam.PcoDetectorCam ophyd.areadetector.cam.PcoDetectorIO ophyd.areadetector.cam.PcoDetectorSimIO ophyd.areadetector.cam.PerkinElmerDetectorCam ophyd.areadetector.cam.PilatusDetectorCam ophyd.areadetector.cam.PixiradDetectorCam ophyd.areadetector.cam.PointGreyDetectorCam ophyd.areadetector.cam.ProsilicaDetectorCam ophyd.areadetector.cam.PvcamDetectorCam ophyd.areadetector.cam.RoperDetectorCam ophyd.areadetector.cam.SimDetectorCam ophyd.areadetector.cam.URLDetectorCam
    :parts: 1
 
+Custom Devices
+==============
+For custom hardware based on area-detector it may be necesary to add a custom
+device class (for custom plugins see section below). The new class should
+inherit from ``ophyd.areadetector.base.ADbase`` and should have the following
+PV structure:
+
+.. code-block:: python
+
+    PV = Areadetector_device_PV_prefix:(Plugin_suffix or attribute_suffix)
+
+As an example, for the builtin areadetector 'stats' class this looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:Stats
+
+And for the builtin areadetector 'color mode' attribute it looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:ColorMode_RBV
+
+where ``Areadetector_device_PV_suffix`` is the PV name for the Area detector
+device, ``plugin_suffix = Stats`` is the 'stats' Plugin suffix and
+``attribute_suffix = ColorMode_RBV`` is the'color mode' attribute suffix.
+
+In order to create the class then the following code is required (where ``XXX``
+is the name of the device):
+
+.. code-block:: python
+
+    from ophyd.areadetector.base import ADComponent as ADCpt, ad_group,
+                                        EpicsSignalWithRBV
+    from ophyd.signal import EpicsSignal, EpicsSignalRO
+    from ophyd.device import DynamicDeviceComponent as DDCpt
+    from ophyd.detectors import DetectorBase
+
+    Class XXX('optional trigger mixin, 'DetectorBase):
```

Comment:
Are the quotes meant to emphasize the optional-ness? It would be better to make this copy/paste-able valid Python that people can iterate from; probably best to include `SingleTrigger` and remark in a comment that it is optional/modular.

Reply:
yes that was my aim, but I see your point on copy/paste. I will make the change you suggest